packages/aws-cdk-lib/aws-cloudfront/lib/distribution.ts

```diff
@@ -787,6 +794,15 @@ export class Distribution extends Resource implements IDistribution {
       sslSupportMethod: sslSupportMethod,
     };
   }
+
+  private validateGrpc(cacheBehavior: CacheBehavior) {
+    if (!cacheBehavior.grpcEnabled) {
+      return;
+    }
+    if (![HttpVersion.HTTP2, HttpVersion.HTTP2_AND_3].includes(this.httpVersion)) {
+      throw new Error(`'httpVersion' must be HttpVersion.HTTP2 or HttpVersion.HTTP2_AND_3 if 'enableGrpc' in 'defaultBehavior' or 'additionalBehaviors' is true, got ${this.httpVersion}`);
+    }
```

Comment:
Otherwise CFn will fail:

```
Resource handler returned message: "Invalid request provided: AWS::CloudFront::Distribution: The parameter gRPC only supports HTTP/2. (Service: CloudFront, Status Code: 400, Request ID:...
```